library/aesce.c

```diff
@@ -0,0 +1,50 @@
+/*
+ *  Arm64 crypto engine support functions
+ *
+ *  Copyright The Mbed TLS Contributors
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#include <string.h>
+#include "common.h"
+
+#if defined(MBEDTLS_AESCE_C)
+
+#include "aesce.h"
+
+#if defined(MBEDTLS_HAVE_ARM64)
+
+#if defined(__clang__)
+#   if __clang_major__ < 4
+#       error "A more recent Clang is required for MBEDTLS_AES_C"
+#   endif
+#elif defined(__GNUC__)
+#   if __GNUC__ < 6
+#       error "A more recent GCC is required for MBEDTLS_AES_C"
+#   endif
+#else
+#    error "Only GCC and Clang supported for MBEDTLS_AES_C"
+#endif
```

Comment:
How did you determine that we need these versions?

(As far as I can tell GCC has ACLE support in GCC 5.1 already:
https://gcc.gnu.org/onlinedocs/gcc-5.1.0/gcc/ARM-C-Language-Extensions-_0028ACLE_0029.html#ARM-C-Language-Extensions-_0028ACLE_0029)

Reply:
I tried each version. For 5.X, I remember some intrinsic functions are miss . Let me double check that.

Reply:
I tried gcc-5.0, and below function are missed .
- vreinterpretq_p64_u8
- vreinterpretq_u8_p128
- vget_low_p64

And I did not tried 5.1, I will try it and update this paragraph

Reply:
- gcc 5.1 5.2, crypto extension can not be enabled with `arch=armv8-a+crypto`
- 5.3 5.4 5.5: some functions are not defined.
   - vreinterpretq_p64_u8, vget_low_p64 vreinterpretq_u8_p128

And there are 5 release version for gcc-5 . So I removed gcc 5 supported.

Reply:
I found a typo problem in this paragraph and  update